scripts/pullsAndImpacts.py

```diff
@@ -0,0 +1,859 @@
+    ncols = pulls + impacts
+    fig = make_subplots(rows=1, cols=ncols, horizontal_spacing=0.1, shared_yaxes=True)
+
+    loffset = 40
+    if title is not None:
+        if subtitle is not None:
+            loffset += max(len(subtitle), len(title)) * 7
+        else:
+            loffset += len(title) * 6
+
+    if legend_pos == "bottom":
+        legend = dict(
+            orientation="h",
+            xanchor="left",
+            yanchor="top",
+            x=0.0,
+            y=0.0,
+        )
+    elif legend_pos == "right":
+        legend = dict(
+            orientation="v",
+            xanchor="left",
+            yanchor="top",
+            x=1.0,
+            y=1.0,
+        )
+    else:
+        raise NotImplementedError("Supported legend positions are ['bottom', 'left']")
+
+    ndisplay = len(df)
+    fig.update_layout(
+        paper_bgcolor="rgba(0,0,0,0)",
+        plot_bgcolor="rgba(0,0,0,0)",
+        xaxis_title=impact_title if impacts else "Pull",
+        margin=dict(l=loffset, r=20, t=50, b=20),
+        yaxis=dict(range=[-1, ndisplay]),
+        showlegend=show_legend,
+        legend=legend,
+        legend_itemsizing="constant",
+        height=100 * (ndisplay < 100)
+        + ndisplay * 20.5
+        + show_legend
+        * (legend_pos != "right")
+        * (impacts + pulls + asym_pulls)
+        * (1 + include_ref)
+        * 25,
+        width=800 if show_legend and legend_pos == "right" else 640,
+        font=dict(
+            color="black",
+        ),
+    )
+
+    gridargs = dict(
+        showgrid=True,
+        gridwidth=1,
+        gridcolor="Gray",
+        griddash="dash",
+        zeroline=True,
+        zerolinewidth=2,
+        zerolinecolor="Gray",
+    )
+    tickargs = dict(
+        tick0=0.0,
+        tickmode="linear",
+        tickangle=0,
+        side="top",
+    )
+
+    text_on_bars = False
+    labels = df["label"]
+    if impacts and show_numbers:
+        if include_ref:
+            # append numerical values of impacts on nuisance name; fill up empty room with spaces to align numbers
+            frmt = (
+                "{:0"
+                + str(
+                    int(
+                        np.log10(max(df["absimpact"]))
+                        if max(df[f"absimpact_ref"]) > 0
+                        else 0
+                    )
+                    + 2
+                )
+                + ".2f}"
+            )
+            nval = df["absimpact"].apply(
+                lambda x, frmt=frmt: frmt.format(x)
+            )  # .astype(str)
+            nspace = nval.apply(
+                lambda x, n=nval.apply(len).max(): " " * (n - len(x) + 1)
+            )
+            if include_ref:
+                frmt_ref = (
+                    "{:0"
+                    + str(
+                        int(
+                            np.log10(max(df[f"absimpact_ref"]))
+                            if max(df[f"absimpact_ref"]) > 0
```

Comment:
I think the thing that is making this fail is a logic error here: in the case where the maximum value is much smaller than 1 but non-zero, the log gives a negative number. So I think it would be good enough to make this > 1, for example, but then I'm not 100% sure I follow the logic of adding 2.